### ISSUE TITLE
Breaking Change: Improve syntax highlighting for consistency with other Catppuccin themes

### ISSUE THREAD

**Issue by nekowinston** [catppuccin/vscode#62]
I have started work on this in #53, creating this issue for more visibility. I aim to release it within a next MINOR update, which would be 2.3.0 or 2.4.0

Catppuccin still doesn't really have a strict styleguide, if we're talking about "Editor styleguides".
The terminal colours are well-defined (even though they're due for a minor rework as well), but for editors we essentially looked to [nvim](https://github.com/catppuccin/nvim) as our guide. Some ports come pretty close, others (like this port, VSCode), miss the mark completely.

@sgoudham and I recently worked on [JetBrains](https://github.com/catppuccin/jetbrains), and while working on it we:
- looked at nvim, took what we could from there
- found flaws in how *unmodified TreeSitter* parses certain Languages (especially JS/TS)
- improved things, while not introducing radical changes.

The first comment in #53 shows my current progress adding the **Python** changes back into VSCode from JetBrains.
I will continue to work on other languages, rough order of my priorities:
- JS/JSX & TS/TSX
- HTML & CSS
- Rust
- Golang
- Everything else

<hr/> 

**TLDR**: This will introduce large changes to the syntax highlighting. IMO our VSCode highlighting is *all but consistent*, and looks straight up terrible in some places.
